here/traffic/sql/create_tables.sql

```diff
@@ -1,73 +1,72 @@
-/*DROP TABLE IF EXISTS here.ta CASCADE;
-
-CREATE TABLE here.ta
+-- Parent table structure
+CREATE TABLE IF NOT EXISTS here.ta
 (
-  link_dir text NOT NULL,
-  tx timestamp without time zone NOT NULL,
-  epoch_min integer NOT NULL,
-  length int,
-  mean numeric(4,1) NOT NULL,
-  stddev numeric(4,1) NOT NULL,
-  min_spd integer NOT NULL,
-  max_spd integer NOT NULL,
-  confidence integer NOT NULL,
-  pct_5 int NOT NULL,
-pct_10 int NOT NULL,
-pct_15 int NOT NULL,
-pct_20 int NOT NULL,
-pct_25 int NOT NULL,
-pct_30 int NOT NULL,
-pct_35 int NOT NULL,
-pct_40 int NOT NULL,
-pct_45 int NOT NULL,
-pct_50 int NOT NULL,
-pct_55 int NOT NULL,
-pct_60 int NOT NULL,
-pct_65 int NOT NULL,
-pct_70 int NOT NULL,
-pct_75 int NOT NULL,
-pct_80 int NOT NULL,
-pct_85 int NOT NULL,
-pct_90 int NOT NULL,
-pct_95 int NOT NULL
-)
-WITH (
-  OIDS=FALSE
-);
+    link_dir text NOT NULL,
+	dt date NOT NULL,
+	tod time without time zone NOT NULL,
+	tx timestamp without time zone NOT NULL, 
+    length integer,
+    mean numeric(4,1) NOT NULL,
+    stddev numeric(4,1) NOT NULL,
+    min_spd integer NOT NULL,
+    max_spd integer NOT NULL,
+    confidence integer NOT NULL,
+	sample_size integer
+) PARTITION BY RANGE (dt);
+
 ALTER TABLE here.ta
   OWNER TO here_admins;
-*/
 
-/*Loops through the years and then months for which we currently have data in order to create a partitioned table for each month*/
+-- simple view structure
+CREATE VIEW here.ta_view AS
+
+SELECT *
+FROM here.ta;
+
+ALTER TABLE here.ta_view
+  OWNER TO here_admins;
+  
+-- Instead of inserting to ta_view, 
+-- insert to parent table trigger
+
+CREATE TRIGGER transform_trigger
+        INSTEAD OF INSERT
+        ON here.ta_view
+        FOR EACH ROW
+        EXECUTE PROCEDURE here.here_insert_trigger();
 
+
+--Loops through the years and then months for which we currently have data in order to create a partitioned table for each month
 DO $do$
```

Comment:
Instead of doing this looping you could loop over that yearly create tables function.

Or use the eoy DAG and run it for 2012 to 2022